### PR TITLE
Use docker container directly in CI

### PR DESCRIPTION
**Pull request by Gamenot**
This does not add much to the size but makes the image more useful.

### CODE REVIEW

File: .github/workflows/ci-base-tests.yml [hunk 1]
```diff
@@ -2,12 +2,10 @@ name: SMARTS CI
 
 on: [push]
 
-env:
-  venv_dir: .venv
```
Comment (JVillella) [inline .github/workflows/ci-base-tests.yml:6]: What about PRs that add a new package? Looks like a docker build will need to be deployed first...but a new package needs to go out before we'd deploy a new build...

Sounds like a 🐔 and 🥚

File: Dockerfile [hunk 2]
```diff
@@ -57,6 +57,10 @@ RUN echo "Cleaning-up"
 RUN apt-get autoremove && \
     rm -rf /var/lib/apt/lists/*
 
+RUN add-apt-repository ppa:git-core/ppa
+RUN apt-get update
+RUN apt-get install -y git
```
Comment (JVillella) [inline Dockerfile:62]: For convenience? What's the use case?

**Review by JVillella — CHANGES_REQUESTED** [review]
See inline comments

File: Dockerfile (hunk 2 above)
Comment (Gamenot) [reply Dockerfile:62]: This is mainly for convenience for development. Perhaps removing the `apt-get install` would be far enough, the docker instance does not have git and `apt` does not know where it is located by default.

File: .github/workflows/ci-base-tests.yml (hunk 1 above)
Comment (Gamenot) [reply .github/workflows/ci-base-tests.yml:6]: That would be the case but we run an update to the packages already in in the scripts

https://github.com/huawei-noah/SMARTS/blob/master/.github/workflows/ci-base-tests.yml#L26-L29

**Review by liamchzh — APPROVED** [review]

File: .github/workflows/ci-base-tests.yml [hunk 3]
```diff
@@ -21,15 +19,12 @@ jobs:
             -logfile ./xdummy.log \
             -config /etc/X11/xorg.conf :1 &
           cd $GITHUB_WORKSPACE
-          python3.7 -m venv "${venv_dir}"
-          . "${venv_dir}/bin/activate"
           pip install --upgrade pip
           pip install wheel
-          pip install -r requirements.txt
+          pip install --no-deps -r requirements.txt
```
Comment (JVillella) [inline .github/workflows/ci-base-tests.yml:24]: why?

if they're cached it'll just go over em quickly

**Review by JVillella — APPROVED** [review]
Small comment, otherwise 👍

File: .github/workflows/ci-base-tests.yml (hunk 3 above)
Comment (Gamenot) [reply .github/workflows/ci-base-tests.yml:24]: I was getting dependency collisions.

File: Dockerfile (hunk 2 above)
Comment (Gamenot) [reply Dockerfile]: This was removed because it was not needed.

**Comment by Gamenot** [comment]
This has suddenly stopped working. Not suitable to go in as is.

File: .github/workflows/ci-base-tests.yml (hunk 3 above)
Comment (JVillella) [reply .github/workflows/ci-base-tests.yml:24]: This is likely _not_ the solution to that. If these are dep collisions of our own dependencies than we need to ease up our requirements.txt / setup.py to let them resolve.

**Review by JVillella — CHANGES_REQUESTED** [review]
Have an issue w/ the --no-deps usage.

**Comment by Gamenot** [comment]
https://github.com/huawei-noah/SMARTS/pull/587 covers this in a better way.

**Pull request closed without merging**